diff -ur o/glest-source-3.1.0/glest_game/facilities/logger.cpp p/glest-source-3.1.0/glest_game/facilities/logger.cpp
--- o/glest-source-3.1.0/glest_game/facilities/logger.cpp	2008-02-19 23:12:27.650654417 +0100
+++ p/glest-source-3.1.0/glest_game/facilities/logger.cpp	2008-02-19 23:29:59.580561861 +0100
@@ -17,6 +17,7 @@
 #include "metrics.h"
 #include "lang.h"
 #include "leak_dumper.h"
+#include <sys/stat.h>
 
 using namespace std;
 using namespace Shared::Graphics;
@@ -46,7 +47,9 @@
 }
 
 void Logger::add(const string &str,  bool renderScreen){
-	FILE *f=fopen(fileName.c_str(), "at+");
+	char logfile[PATH_MAX];
+	snprintf(logfile, PATH_MAX, "%s/.glest/%s", getenv("HOME"), fileName.c_str());
+	FILE *f=fopen(logfile, "at+");
 	if(f==NULL){
 		throw runtime_error("Error opening log file"+ fileName);
 	}
@@ -77,7 +80,10 @@
 void Logger::clear(){
     string s="Log file\n";
 
-	FILE *f= fopen(fileName.c_str(), "wt+");
+        char logfile[PATH_MAX];
+        snprintf(logfile, PATH_MAX, "%s/.glest/%s", getenv("HOME"), fileName.c_str());
+
+	FILE *f= fopen(logfile, "wt+");
 	if(f==NULL){
 		throw runtime_error("Error opening log file"+ fileName);
 	}
diff -ur o/glest-source-3.1.0/glest_game/main/main.cpp p/glest-source-3.1.0/glest_game/main/main.cpp
--- o/glest-source-3.1.0/glest_game/main/main.cpp	2008-02-19 22:36:13.000000000 +0100
+++ p/glest-source-3.1.0/glest_game/main/main.cpp	2008-02-19 22:03:18.000000000 +0100
@@ -13,6 +13,7 @@
 
 #include <string>
 #include <cstdlib>
+#include <sys/stat.h>
 
 #include "game.h"
 #include "main_menu.h"
@@ -133,6 +134,17 @@
 	ExceptionHandler exceptionHandler;
 	exceptionHandler.install();
 
+	if (!getenv("HOME"))
+		throw runtime_error("HOME not set!");
+
+	char str[PATH_MAX];
+	snprintf(str, PATH_MAX, "%s/.glest", getenv("HOME"));
+	mkdir(str, 0750);
+	chdir(str);
+	mkdir("screens", 0750);
+
+	chdir("GENTOO_DATADIR");
+
 	try{
 		Config &config = Config::getInstance();
 
diff -ur o/glest-source-3.1.0/shared_lib/sources/util/properties.cpp p/glest-source-3.1.0/shared_lib/sources/util/properties.cpp
--- o/glest-source-3.1.0/shared_lib/sources/util/properties.cpp	2008-02-08 11:43:47.000000000 +0100
+++ p/glest-source-3.1.0/shared_lib/sources/util/properties.cpp	2008-02-19 22:05:22.000000000 +0100
@@ -33,9 +33,13 @@
 
 	this->path= path;
 
-	fileStream.open(path.c_str(), ios_base::in);
+	char str[PATH_MAX];
+	snprintf(str, PATH_MAX, "%s/.glest/%s", getenv("HOME"), path.c_str());
+	fileStream.open(str, ios_base::in);
 	if(fileStream.fail()){
-		throw runtime_error("Can't open propertyMap file: " + path);
+		fileStream.open(path.c_str(), ios_base::in); // use defaults
+		if(fileStream.fail())
+			throw runtime_error("Can't open properties file: " + path);
 	}
 
 	propertyMap.clear();
@@ -71,7 +75,9 @@
 void Properties::save(const string &path){
 	ofstream fileStream;
 
-	fileStream.open(path.c_str(), ios_base::out | ios_base::trunc);
+	char str[PATH_MAX];
+	snprintf(str, PATH_MAX, "%s/.glest/%s", getenv("HOME"), path.c_str());
+	fileStream.open(str, ios_base::out | ios_base::trunc);
 
 	fileStream << "; === propertyMap File === \n";	
 	fileStream << '\n';
