--- src/NetworkManager.conf.bak	2008-04-24 12:35:07.000000000 +0200
+++ src/NetworkManager.conf	2008-04-24 12:36:53.000000000 +0200
@@ -11,18 +11,18 @@
                 <allow send_destination="org.freedesktop.NetworkManager.PPP"/>
                 <allow send_interface="org.freedesktop.NetworkManager.PPP"/>
         </policy>
-        <policy at_console="true">
+        <policy group="plugdev">
                 <allow send_destination="org.freedesktop.NetworkManager"/>
                 <allow send_interface="org.freedesktop.NetworkManager"/>
         </policy>
         <policy context="default">
-                <deny own="org.freedesktop.NetworkManager"/>
-                <deny send_destination="org.freedesktop.NetworkManager"/>
-                <deny send_interface="org.freedesktop.NetworkManager"/>
+                <allow own="org.freedesktop.NetworkManager"/>
+                <allow send_destination="org.freedesktop.NetworkManager"/>
+                <allow send_interface="org.freedesktop.NetworkManager"/>
 
-                <deny own="org.freedesktop.NetworkManager.PPP"/>
-                <deny send_destination="org.freedesktop.NetworkManager.PPP"/>
-                <deny send_interface="org.freedesktop.NetworkManager.PPP"/>
+                <allow own="org.freedesktop.NetworkManager.PPP"/>
+                <allow send_destination="org.freedesktop.NetworkManager.PPP"/>
+                <allow send_interface="org.freedesktop.NetworkManager.PPP"/>
         </policy>
 
         <limit name="max_replies_per_connection">512</limit>
