Index: compress-man.bash
===================================================================
--- compress-man-1.2/compress-man.bash	(revision 649)
+++ compress-man-1.2/compress-man.bash	(working copy)
@@ -6,6 +6,7 @@
    ewarn "Cannot compress man and info files, symlinks not found to fix symbolic links"
    exit 0
 fi
+SYMLINKS=$(which symlinks)
 
 if CMD=`which bzip2 2> /dev/null`; then
    EXT="bz2"
