--- src/wordlist.c.orig	2006-02-27 04:46:56.000000000 +0100
+++ src/wordlist.c	2008-12-09 12:21:48.000000000 +0100
@@ -102,7 +102,7 @@
 
 	if (word_file == stdin) return -1;
 
-	if (fstat(fileno(word_file), &file_stat)) pexit("fstat");
+	if (fstat64(fileno(word_file), &file_stat)) pexit("fstat64");
 
 	if ((pos = ftell(word_file)) < 0) {
 #ifdef __DJGPP__
@@ -137,8 +137,8 @@
 	log_event("Proceeding with wordlist mode");
 
 	if (name) {
-		if (!(word_file = fopen(path_expand(name), "r")))
-			pexit("fopen: %s", path_expand(name));
+		if (!(word_file = fopen64(path_expand(name), "r")))
+			pexit("fopen64: %s", path_expand(name));
 		log_event("- Wordlist file: %.100s", path_expand(name));
 	} else {
